pangea-sdk/pangea/pangea.go

```diff
@@ -213,51 +224,38 @@ func mergeHeaders(req *http.Request, additionalHeaders map[string]string) {
 	}
 }
 
-func (c *Client) GetRequestIDURL(rid string) (string, error) {
+func (c *Client) GetRequestIDURL(rid string) (*url.URL, error) {
 	return c.GetURL(fmt.Sprintf("request/%v", rid))
 }
 
-func (c *Client) GetURL(path string) (string, error) {
-	cfg := c.config
-	endpoint := ""
-	// Remove slashes, just in case
-	path = strings.TrimPrefix(path, "/")
-	domain := strings.TrimSuffix(cfg.Domain, "/")
-
-	if strings.HasPrefix(cfg.Domain, "http://") || strings.HasPrefix(cfg.Domain, "https://") {
-		// URL
-		endpoint = fmt.Sprintf("%s/%s", domain, path)
-	} else {
-		scheme := "https://"
-		if cfg.Insecure {
-			scheme = "http://"
-		}
-		if cfg.Environment == "local" {
-			// If we are testing locally do not use service
-			endpoint = fmt.Sprintf("%s%s/%s", scheme, domain, path)
-		} else {
-			endpoint = fmt.Sprintf("%s%s.%s/%s", scheme, c.serviceName, domain, path)
-		}
+func (c *Client) GetURL(path string) (*url.URL, error) {
+	u, err := url.Parse(strings.Replace(c.config.BaseURLTemplate, serviceNamePlaceholder, c.serviceName, 1))
```

Comment:
This could be replaced and validated just once when initialising the service, it should not change over calls

Reply:
What held me back from doing that is that parsing the URL can return an `error`, but functions like `NewClient()` don't currently return `error`s. So, it ends up requiring more refactoring.